src/caffe/layers/cudnn_conv_layer.cu

```diff
@@ -19,6 +19,11 @@ void CuDNNConvolutionLayer<Dtype>::Forward_gpu(
     Dtype* top_data = top[i]->mutable_gpu_data();
     const Dtype* weight = this->blobs_[0]->gpu_data();
 
+    size_t workspace_limit_bytes = this->kernel_h_ *
+                                   this->kernel_w_ *
+                                   this->channels_ *
+                                   sizeof(int) + 1;
```

Comment:
It seems odd to me that the workspace can be the same size whether the input and filters are float or double, but if that's how it is in cuDNN then so be it.